pkg/oc/cli/deploymentconfigs/delete.go

```diff
@@ -18,14 +19,15 @@ import (
 )
 
 // NewDeploymentConfigReaper returns a new reaper for deploymentConfigs
-func NewDeploymentConfigReaper(appsClient appsclient.Interface, kc kclientset.Interface) kubectl.Reaper {
-	return &DeploymentConfigReaper{appsClient: appsClient, kc: kc, pollInterval: kubectl.Interval, timeout: kubectl.Timeout}
+func NewDeploymentConfigReaper(appsClient appsclient.Interface, kc kclientset.Interface, scaleClient scaleclient.ScalesGetter) kubectl.Reaper {
```

Comment:
@mfojtik can we finally drop the reapers? we have GC since 3.6. We should do that upstream as well.

Reply:
We can't entirely drop the reapers, but I need to start doing this upstream. So upstream first gets rid of them and then we'll do it during after the next rebase. That reminds me I need to sync with @deads2k (be prepared my list of topics is growing :wink:).

Reply:
> @soltysh I am interested in those reasons

Reaper aside from removing a resource also scale them down, eg. deployments. That bit of functionality cannot be removed and I doubt the GC is handling that.

Reply:
@soltysh isn't the resource marked as deleted so it can't create new subresources? No need for scale down...

Reply:
That's something I'm not 100% sure of and as usual I'd prefer to double check it.

Reply:
Controllers are not allowed to operate on resources with `deletionTimestamp != nil` to avoid racing with GC

Reply:
Yup.

Reply:
I am quite positive but double checking doesn't hurt :)